src/Statement.py

```diff
@@ -24,45 +24,54 @@ def get_category(self):
         """
         Check the category of the given statement.
 
-        :return: category of the statement; if unknown return the empty string
+        :return: category of the statement; if ignored on purpose return 'Ignored', if unknown return the empty string
         """
         booking_type = self._statement[self._config.get_booking_type()]
-        value = self.get_value()
         category = ""
-        if self._config.get_relevant_income_regex().match(booking_type):
-            category = "Zinsen"
-        elif self._config.get_relevant_invest_regex().match(booking_type):
-            category = "Einlage"
-        elif self._config.get_relevant_payment_regex().match(booking_type):
-            category = "Entnahme"
-        elif self.is_fee(booking_type):
-            category = "Gebühren"
-        elif self._config.get_special_entry_regex() and self._config.get_special_entry_regex().match(booking_type):
-            if value > 0:
-                category = "Zinsen"
-            elif value < 0:
-                category = "Gebühren"
-            else:
-                logging.debug(self._statement)
-        elif self._config.get_ignorable_entry_regex() and self._config.get_ignorable_entry_regex().match(booking_type):
-            category = "Ignored"
-        else:
-            logging.debug(self._statement)
+
+        regex_to_category_mappings = [
+            {"regex": self._config.get_relevant_income_regex(), "category": "Zinsen"},
+            {"regex": self._config.get_relevant_invest_regex(), "category": "Einlage"},
+            {"regex": self._config.get_relevant_payment_regex(), "category": "Entnahme"},
+            {"regex": self._config.get_relevant_fee_regex(), "category": "Gebühren"},
+            {"regex": self._config.get_special_entry_regex(), "category": "Undecided"},
+            {"regex": self._config.get_ignorable_entry_regex(), "category": "Ignored"},
+        ]
+
+        for mapping in regex_to_category_mappings:
+            if mapping["regex"] and mapping["regex"].match(booking_type):
+                category = mapping["category"]
+                break
+
+        if category == "Undecided":
+            category = Statement.__handle_special_case_mintos_discount_premium(self.get_value())
+
+        if category == "":
```

Comment:
```suggestion
        if not category:
```

empty string is handled as Boolean False :)

Reply:
Will make sure to remove all these comparisons with "" :)